### DISCUSSION TITLE
Bug when using x-trap and input of type "date"

### DISCUSSION THREAD

**Discussion by dlebedef** [alpinejs/alpine#4213] [7. Bugs (archived)]
I have a little modal with two input dates in it and when I apply the `x-trap` attribute I have a weird and unexpected behaviour.
Normally I can TAB through the '**dd**', '**mm**' and '**yyyy**' parts of the input date element but this behaviour changes if the input date is the last item in the modal, in that case it will only focus on the 'dd' part and then restart the focus selection on the first focusable item of the active modal.

Here is a basic example of the html structure:

```html
<div x-data="{open: false}">
    <button @click="open = !open">Open</button>
     <div x-show="open" x-trap="open">
        <div><input type="text" name="" id="input-text"></div>
        <div><input type="date" name="" id="input-date-1"></div>
        <div><input type="date" name="" id="input-date-2"></div>
    </div>
</div>
```

If I add any other focusable element after the last input date everything works fine, this only happens if the input date is the last focusable element in the modal.

Any idea on how to solve this ?

**Comment by ekwoka** [comment]
So this is a bug...

Not sure if there is an easy fix...

Basically, since the input is the last focusable, the control is hijacked and moved to the front.

But the native behavior is that it would actually stay inside the date...

:thinking:

[reference sandbox](https://awesomealpine.com/play/?coreplugins=4&html=PGRpdgogIHgtZGF0YT0ie29wZW46IGZhbHNlfSIKICBjbGFzcz0iZmxleCBmbGV4LWNvbCBpdGVtcy1jZW50ZXIganVzdGlmeS1jZW50ZXIgZ2FwLTQiCj4KICA8YnV0dG9uCiAgICBAY2xpY2s9Im9wZW4gPSAhb3BlbiIKICAgIGNsYXNzPSJweC00IHB5LTIgYmctaW5kaWdvLTcwMCB0ZXh0LXdoaXRlIHJvdW5kZWQiCiAgPgogICAgT3BlbgogIDwvYnV0dG9uPgogIDxkaXYKICAgIHgtc2hvdz0ib3BlbiIKICAgIHgtdHJhcD0ib3BlbiIKICAgIGNsYXNzPSJmbGV4IGZsZXgtY29sIGdhcC0yIGl0ZW1zLWNlbnRlciBiZy13aGl0ZSBzaGFkb3cgcm91bmRlZC1tZCBwLTQiCiAgPgogICAgPGRpdj4KICAgICAgPGlucHV0CiAgICAgICAgdHlwZT0idGV4dCIKICAgICAgICBuYW1lPSIiCiAgICAgICAgaWQ9ImlucHV0LXRleHQiCiAgICAgIC8-CiAgICA8L2Rpdj4KICAgIDxkaXY-CiAgICAgIDxpbnB1dAogICAgICAgIHR5cGU9ImRhdGUiCiAgICAgICAgbmFtZT0iIgogICAgICAgIGlkPSJpbnB1dC1kYXRlLTEiCiAgICAgIC8-CiAgICA8L2Rpdj4KICAgIDxkaXY-CiAgICAgIDxpbnB1dAogICAgICAgIHR5cGU9ImRhdGUiCiAgICAgICAgbmFtZT0iIgogICAgICAgIGlkPSJpbnB1dC1kYXRlLTIiCiAgICAgIC8-CiAgICA8L2Rpdj4KICA8L2Rpdj4KPC9kaXY-Cg&script=Y29uc29sZS5sb2coJ2hlbGxvJyk7Cg&tw=1)
